activerecord/lib/rails/generators/active_record/model/model_generator.rb

```diff
@@ -43,9 +43,16 @@ def accessible_attributes
 
         # Used by the migration template to determine the parent name of the model
         def parent_class_name
-          options[:parent] || "ActiveRecord::Base"
+          options[:parent] || determine_default_parent_class
         end
 
+        def determine_default_parent_class
+          if File.exist?('app/models/application_record.rb')
```

Comment:
I made the check to be relative to the application root here https://github.com/rails/rails/commit/342221b073c82ac7ebb7349da8b1d7e1ab1c4005